src/Storages/StorageS3.cpp

```diff
@@ -794,7 +794,7 @@ class StorageS3Sink : public SinkToStorage
     void onException() override
     {
         std::lock_guard lock(cancel_mutex);
-        finalize();
+        abort();
```

Comment:
Ok. I was scared reading this function name, because it is the same as C's abort. Maybe `cancel`?

Reply:
`void onCancel()` is already in use.
Actually it is better to fix `onCancel` as well. But I do not have a test for it. And it is not urgent at all if compare to the changes in `onException`.

Reply:
May be `release` is fine here.